aptos-move/framework/aptos-token-objects/sources/collection.move

```diff
@@ -97,16 +105,39 @@ module aptos_token_objects::collection {
         mint_events: event::EventHandle<MintEvent>,
     }
 
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Supply tracker, useful for tracking amount of issued tokens.
+    /// If max_value is not set to U64_MAX, this ensures that a limited number of tokens are minted.
+    struct ConcurrentSupply has key {
+        /// Total minted - total burned
+        current_supply: Aggregator<u64>,
+        total_minted: Aggregator<u64>,
+    }
+
     struct BurnEvent has drop, store {
         index: u64,
         token: address,
     }
 
+    #[event]
+    struct ConcurrentBurnEvent has drop, store {
+        collection_addr: address,
+        index: AggregatorSnapshot<u64>,
```

Comment:
it depends on how many transactions have minted before you - basically index here needs to be exactly in the order in which transactions are ordered in the block.

and even more convoluted, we can reach the collection limit, then some transactions burn a few tokens, and then first ones coming after that can mint successfully.

Reply:
Did I get it wrong? This is `ConcurrentBurnEvent` and index is a deterministic argument passed in from the caller. Why do we need a snapshot here...

Reply:
oh burn event. you are right

Reply:
But you ask a very interesting question: mint, burn, and remint... I don't think we want to support that because I have no idea how we assign the right index for remint.

Reply:
that's already supported :)

we have two counters, index is based on minting order, and supply limits if the new token can be minted

Reply:
My question is if we have 1-10 minted and then burn 6. Could 6 be reminted with the right index?

Reply:
No, index only increases.